editors/vscode/slint.tmLanguage.json

```diff
@@ -13,32 +13,153 @@
                     "include": "#general"
                 },
                 {
-                    "match": "\\b(import|from|export|struct|enum|global|component|inherits)\\b",
+                    "match": "(?<!-)\\b(from|export)\\b(?!-)",
                     "name": "keyword"
                 },
                 {
-                    "match": "\\b(property|callback|animate|states|transitions|private|public|pure|function|in|out|in-out)\\b",
+                    "match": "(?<!-)\\b(animate|states|transitions|private|public|pure|function|in|out|in-out)\\b(?!-)",
                     "name": "keyword.other"
                 },
                 {
-                    "match": "\\b(if|for|return)\\b",
+                    "match": "(?<!-)\\b(if|for|return)\\b(?!-)",
                     "name": "keyword.control"
                 },
                 {
-                    "match": "\\b(root|parent|this|self)\\b",
+                    "match": "(?<!-)\\b(root|parent|this|self)\\b(?!-)",
                     "name": "variable.language"
                 },
                 {
-                    "match": "[a-zA-Z_][a-zA-Z_\\-0-9]* *:",
-                    "name": "variable.parameter"
+                    "match": "(?<!-)\\b(import)\\s*\\{\\s*(([a-zA-Z_][a-zA-Z0-9_-]*)(\\s*,\\s*([a-zA-Z_][a-zA-Z0-9_-]*))*)\\s*\\}",
```

Comment:
not sure it is so wise to try to do parsing in there.  I think this misses the  `import { Foo as Bar }` for example.
If you do, you can also include "from" because it can only appear after an import.

Note that Slint don't really have keywords anyway.
And we could distinguish between keywords that can appear at the document level (outside of any scope) like import, export, component, global, struct, enum, from, inherits.
And the keyword that appear within components